File: DuDoong-Api/src/main/java/band/gosrock/api/host/service/ReadHostUseCase.java
package band.gosrock.api.host.service;


import band.gosrock.api.host.model.dto.response.HostDetailResponse;
import band.gosrock.api.host.model.mapper.HostMapper;
import band.gosrock.common.annotation.UseCase;
import lombok.RequiredArgsConstructor;
import org.springframework.transaction.annotation.Transactional;

@UseCase
@RequiredArgsConstructor
public class ReadHostUseCase {
    private final HostMapper hostMapper;

    @Transactional(readOnly = true)
    public HostDetailResponse execute(Long hostId) {
        return hostMapper.toHostDetailResponse(hostId);
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/host/service/ReadHostUseCase.kt
package band.gosrock.api.host.service

import band.gosrock.api.host.model.dto.response.HostDetailResponse
import band.gosrock.api.host.model.mapper.HostMapper
import band.gosrock.common.annotation.UseCase
import org.springframework.transaction.annotation.Transactional

@UseCase
class ReadHostUseCase(
    private val hostMapper: HostMapper,
) {
    @Transactional(readOnly = true)
    fun execute(hostId: Long): HostDetailResponse {
        return hostMapper.toHostDetailResponse(hostId)
    }
}
